src/components/SubHeader/SubHeader.js

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+
+import { INITIAL_STATE } from "./constants/SubHeader.general";
+import getBreadCrumbConfig from "./helpers/breadCrumbConfig";
+import restaurantReader from "./readers/RestaurantReader";
+
+import { getRestaurantDetails } from "../../apis/getRestaurantDetails";
+
+import Spinner from "../../commonComponents/Spinner";
+import ErrorHandler from "../../commonComponents/ErrorHandler";
+import BreadCrumb from "../../commonComponents/BreadCrumb";
+import Header from "./Restaurant/Header";
+import Feedback from "./Restaurant/Feedback";
+import MenuFilter from "./Restaurant/MenuFilter";
+import Offers from "./Restaurant/Offers";
+import ratingIcon from "../../assets/icons/star.svg";
+
+import "./SubHeader.css";
+
+class SubHeader extends React.Component {
+  state = INITIAL_STATE;
+
+  componentDidMount() {
+    this.fetchRestaurantDetails();
+  }
+
+  fetchRestaurantDetails() {
+    const handleData = function handletData(data) {
+      this.setState({
+        restaurantDetails: data,
+      });
+    }.bind(this);
+
+    const handleError = function handleError(err) {
+      this.setState({
+        errors: {
+          statusCode: 500,
+        },
+      });
+    }.bind(this);
+
+    const handleCleanup = function handleCleanup() {
+      this.setState({
+        isLoading: false,
+      });
+    }.bind(this);
+
+    getRestaurantDetails(1)
+      .then(handleData)
+      .catch(handleError)
+      .finally(handleCleanup);
+  }
+
+  render() {
+    const { isLoading } = this.state;
+
+    if (isLoading) {
+      return <Spinner text="Loading..." />;
+    }
+
+    const { errors } = this.state;
+    if (errors) {
```

Comment:
please check if we need to store errors in state, we might just want to show error to user in a toaster and continue with the same render